### DISCUSSION TITLE
2024-Inthon-Datathon/qna

### DISCUSSION THREAD

**Discussion by giscus** [KU-InThon-Datathon/2024-Inthon-Datathon#7] [Q&A]
# 2024-Inthon-Datathon/qna

A starter template for a Jeykll site using the Just the Docs theme!

https://ku-inthon-datathon.github.io/2024-Inthon-Datathon/qna.html

**Comment by seooyxx** [comment]
대회와 관련 있는 질문을 작성해 주세요.
모든 QnA는 전체 공개로 진행됩니다.

**Comment by PROLCY** [comment]
데이터 전처리에서 외부모델 사용해도 되는지 궁금합니다!

**Reply by KyleBae1017** [reply]
안녕하세요. 이번 데이터톤에서는 오직 Rules 란의 '사용 가능한 모델 목록'에 있는 모델만 사용 가능합니다. 이는 모델 개발의 모든 과정에 해당되므로, 전처리에도 사용이 불가능합니다. 감사합니다.

**Reply by KyleBae1017** [reply]
Q: I am wondering if I can use an external model for data preprocessing!

A: Hello. For this datathon, only the models listed in the "사용 가능한 모델 목록" ("Available Models") section of the Rules are permitted. This restriction applies to all stages of model development, including preprocessing. Thank you.

**Comment by changbeendata** [comment]
안녕하세요, 모델 불러오기 관련하여 질문드립니다. Rules에 있는 6개 모델을 제외한 것은 사용 불가능한 것으로 보이는데, 베이스라인 코드에서 Evaluation 수행 시 사용하는 CLIP 모델을 전처리, 학습, 추론에 사용하는 것도 이 Rule에 위반되는 것인지 여쭈고 싶습니다!

**Reply by KyleBae1017** [reply]
안녕하세요. CLIP은 평가에만 사용되는 모델이고, 참가자 분들께서 사용 가능한 모델이 아닙니다. 참가자 분들의 이해와 편의성을 위해 CLIP을 이용한 채점을 해당 베이스라인 코드에 첨부해둔 것이기에, 이도 위와 마찬가지로 사용 불가능합니다. 즉, **CLIP의 사용이 어떠한 방식으로든 모델의 결과에 영향을 미쳐선 안 됩니다.** 감사합니다.

**Reply by KyleBae1017** [reply]
Q: Hello, I have a question regarding loading models. It seems that only the six models listed in the Rules are allowed, so I wanted to confirm whether using the CLIP model, which is used for evaluation in the baseline code, for preprocessing, training, or inference would violate this Rule.

A: Hello. CLIP is a model designated only for evaluation purposes and is not among the models available for participant use. We have included CLIP-based evaluation in the baseline code for your convenience, but its use is also prohibited. In other words, **CLIP should not influence your model's results in any way.** Thank you.

**Comment by PROLCY** [comment]
10. 평가 및 메트릭 계산 부분 두 번째 셀에
reference_captions = batch['reference_caption']
요부분 에러나는데 혹시 확인해주실 수 있나요??

**Reply by yms020615** [reply]
**7. 커스텀 데이터셋 클래스 및 DataLoader 생성**의 **첫 번째 셀**을 아래와 같이 수정해 주세요.
불편을 드려 죄송합니다.

Q: There is an error in the second cell of 평가 및 메트릭 계산 part,
reference_captions = batch['reference_caption']
Can you check it??
A: Please modify **the first cell** of **7. 커스텀 데이터셋 클래스 및 DataLoader 생성**  as below.
I'm sorry for the inconvenience.

```
from torch.utils.data import Dataset
from PIL import Image
import requests
from io import BytesIO

class CustomImageCaptionDataset(Dataset):
    def __init__(self, df, processor, test=False):
        self.df = df
        self.processor = processor
        self.test = test

    def __len__(self):
        return len(self.df)

    def __getitem__(self, idx):
        if not self.test:
            image_url = self.df.iloc[idx]["url"]
            image_id = self.df.iloc[idx]["Image_ID"]
            caption = self.df.iloc[idx]["Paragraph"]

            response = requests.get(image_url)
            image = Image.open(BytesIO(response.content)).convert('RGB')

            inputs = self.processor(
                images=image,
                text=caption,
                padding="max_length",
                truncation=True,
                max_length=256,
                return_tensors="pt"
            )
            inputs = {k: v.squeeze(0) for k, v in inputs.items()}
            inputs['labels'] = inputs['input_ids'].clone()

            inputs['image_url'] = image_url
            inputs['Image_ID'] = image_id
            inputs['reference_caption'] = caption

        else:
            image_url = self.df.iloc[idx]["url"]
            image_id = self.df.iloc[idx]["Image_ID"]

            inputs = dict()
            inputs['image_url'] = image_url
            inputs['Image_ID'] = image_id

        return inputs
```

**Comment by PROLCY** [comment]
주어진 모델만 사용해서 앙상블 가능한 지 궁금합니다!

**Reply by yms020615** [reply]
가능합니다.

Q: I was wondering if it would be possible to ensemble using only the given model!
A: It is available.

**Comment by dreameerbb** [comment]
가용 모델 중 blip2 모델들이 qformer_tokenizer 파일을 요구하지만 해당 모델 리포지토리 내 파일에 존재하지 않는것 같습니다.

**Reply by yms020615** [reply]
맞습니다. Baseline code는 InstructBLIP을 기반으로 구성되어 있기 때문에 BLIP-2 계열을 사용하고자 하시는 경우 해당 부분을 제외하시거나 수정을 하셔도 괜찮습니다.

Q: Among the available models, the BLIP-2 models seem to require the **qformer_tokenizer** file, but it doesn't appear to be present in the model repository files.

A: That’s correct. The baseline code is structured based on InstructBLIP, so if you wish to use the BLIP-2 series, you can either exclude or modify this part as needed.

**Comment by strn18** [comment]
이미지의 density 값에 대한 자세한 설명을 부탁드려도 될까요? 가령 density 값의 의미나 해당 값을 구하는 공식 등이 궁금합니다.

**Reply by yms020615** [reply]
후원사인 Pebblous 측이 경량화 과정에서 도출된 density를 저희에게 제공해 준 것입니다. 그렇기 때문에 해당 값을 구하는 공식은 저희도 알 수 없지만 유사성이나 중복성을 판단하는 지표는 맞습니다.

아래는 저희가 Pebblous 측에 같은 질문을 문의 드렸을 때 받은 답변입니다.

_경량화의 기준은 density가 맞으며 기준 density를 설정하여 각 단계 별 다이어트를 진행하였습니다._

_density의 계산법은 보안상의 이유로 공개할 수 없다는 점 양해바랍니다. 다만 의미는 말씀해주신 중복성 또는 유사성을 판단하는 지표가 맞습니다._

_계산식이 단순히 질량/부피는 아니기에 절대적인 값으로 사용될 수는 없고 하나의 임베딩 공간안에서 상대적인 척도로 사용된다고 생각해주시면 좋을 것 같습니다._

**Reply by yms020615** [reply]
Q: Could you please provide a detailed explanation of the density value for images? I am curious about the meaning of the density value and the formula used to calculate it.

A: The density value was provided to us by our sponsor, Pebblous, as part of the model's lightweighting process. Therefore, we are not aware of the exact formula used to calculate it, but it does serve as a metric for assessing similarity or redundancy.

Below is the response we received from Pebblous when we inquired about the same question:

_"Density is indeed the standard used for lightweighting, and we conducted incremental optimization based on a set density threshold for each stage."_

_"Due to security reasons, we cannot disclose the exact calculation method for density. However, as you mentioned, it serves as a metric for assessing similarity or redundancy."_

_"The calculation is not simply mass/volume, so it should not be treated as an absolute value. Rather, it is used as a relative measure within a single embedding space."_

**Comment by hiyseo** [comment]
Pebblous에서 각 level이 무엇을 의미하는지 알 수 있을까요?

**Reply by KyleBae1017** [reply]
여러분에게 제공해 드린 데이터는 Level 1이며, 해당 데이터를 단계별로 경량화한 것이 Level 2~11이라고 생각하시면 됩니다. Level이 높을 수록 경량화가 많이 된 데이터라고 생각하시면 됩니다. 경량화는 density를 기준으로 Pebblous 측에서 수행해주셨습니다.

**Reply by KyleBae1017** [reply]
Q: Could you explain what each level in Pebblous means?

A: The data provided to you is Level 1, and the data for Levels 2–11 has been progressively reduced. The higher the level, the more compressed the data. The compression was done by Pebblous based on density.

**Comment by junhwan26** [comment]
데이터 후처리 기준에 대해 설명해주실 수 있나요?

**Reply by KyleBae1017** [reply]
모델이 생성한 결과를 바로 submission으로서 제출해야 한다는 의미입니다. 모델이 생성한 결과에 어떠한 방법으로든 수정을 가하는 것은 모두 후처리로 판단할 예정입니다. 그렇기에 만약 모델 생성 결과가 제출물과 다르다면 이 또한 부정행위에 해당됩니다.

**Reply by KyleBae1017** [reply]
Q: Could you explain the criteria for data post-processing?

A: This means that the results generated by your model should be submitted directly as the submission. Any modification of the model’s output in any way will be considered post-processing. Therefore, if the submitted results differ from the model’s output, this will also be regarded as misconduct.

**Reply by junhwan26** [reply]
앙상블 기법등으로 모델이 생성한 결과들을 처리하는 것 까지 모델로 보는 건가요?

**Reply by yms020615** [reply]
예를 들어 앙상블로 2개의 모델을 사용한다고 했을 때, 1번 모델의 출력을 아무런 처리도 하지 않고 곧바로 2번 모델에 넣는 등의 방법은 괜찮습니다.

또는 1번 모델과 2번 모델의 출력 중 하나를 선택하는 것도 괜찮습니다.

그러나 1번 모델과 2번 모델의 출력을 이어붙이는 것 등의 방법은 최종 결과 자체가 사람에 의해 수정되는 형태이기 때문에 허용되지 않습니다.

**Reply by yms020615** [reply]
Q: Does the processing result generated by models through ensemble methods also count as part of the model?

A: For example, when using two models in an ensemble, it is acceptable to input the output of Model 1 directly into Model 2 without any additional processing. 

It is also acceptable to select one output from either Model 1 or Model 2. 

However, methods such as concatenating the outputs of Model 1 and Model 2 are not permitted, as this would involve human intervention in modifying the final result.

**Reply by PROLCY** [reply]
1번 모델의 출력을 임의로 처리하고 2번 모델에 넣는 방식은 어떨까요??

**Reply by PROLCY** [reply]
임의라는 것은 1번 모델의 출력에 추가적으로 프롬포트를 더해주는 것을 말합니다!

**Reply by yms020615** [reply]
1번 모델의 출력을 직접 처리하는 것은 불가능하지만,
processor의 **text** hyperparameter를 통해서 prompt engineering을 하는 것은 가능합니다.

Q: How about processing the output of Model 1 arbitrarily before feeding it into Model 2? By "arbitrarily," I mean adding an additional prompt to Model 1's output!

A: Directly processing Model 1's output is not allowed, but you can use prompt engineering through the **text** hyperparameter of the processor.

**Comment by junhwan26** [comment]
사용하면 안되는 모델의 기준이 pre-trained 된 모델을 말하는 거 맞나요?

**Reply by yms020615** [reply]
Rules의 **사용 가능한 모델 목록**에 없는 **pre-trained**된 체크포인트를 가져오는 것이 불가능하다는 의미입니다.
즉, 다른 모델의 구조에 대해 from scratch로 학습을 하는 것은 괜찮습니다.

**Reply by yms020615** [reply]
Q: Does the restriction on models mean that pre-trained models are not allowed?

A: It means that bringing in **pre-trained** checkpoints not listed in **사용 가능한 모델 목록** in the Rules is not permitted. However, it is fine to train other model architectures from scratch.

**Comment by moon44432** [comment]
자체적인 추론 및 평가 함수를 만드는 중인데, 평가 및 메트릭 계산의 evaluation_objects 리스트를 calculate_chair_metrics() 함수 외에서 사용이 가능한가요?

**Reply by yms020615** [reply]
가능합니다.

Q: I am creating a custom inference and evaluation function. Is it possible to use the `evaluation_objects` list for evaluation and metric calculation outside of the `calculate_chair_metrics()` function?

A: Yes, it is possible.

**Comment by strn18** [comment]
test set에 대한 채점 시, chair 값은 베이스라인의 동일한 calculate_chair_metrics 함수를 사용해 값을 도출하나요?

**Reply by yms020615** [reply]
네 맞습니다.

Q: When scoring the test set, is the chair value calculated using the same `calculate_chair_metrics` function from the baseline?

A: Yes, that’s correct.

**Comment by PROLCY** [comment]
데이터를 제출할 때 큰따옴표가 있어야 테스트를 할 수 있는 것 같은데, 이를 모델 아웃풋에 추가로 붙이는 것도 안될까요? 모델 아웃풋에 큰따옴표가 포함되어 있어야 하는지 궁금합니다! 자꾸 큰따옴표없이 생성되네요..

**Reply by yms020615** [reply]
질문의 의도를 파악하지 못하여서 조금 더 자세히 설명해 주실 수 있나요?

**Reply by BaconTomatoDeluxe** [reply]
모델을 통해서 submission 생성을 위해서 caption을 생성했을때 caption이 ""없이 생성되는 경우가 있습니다.
e.g)
712e256c,"a red train on a platform at a station with a sign in the window of the front passenger door, and a sign in the window of the back passenger door"
1130bc2a,a man standing in front of a tv playing a bowling game on a nintendo wii video game console

모든 caption이 ""안에 적혀 있어야 제출 확인이 가능한것 같은데 모델이 생성할 때 캡션이 큰따옴표 안에 생성이 되어야하는지, 아니면 큰따옴표를 붙이는 정도의 후처리는 가능한지 궁금합니다.

**Reply by yms020615** [reply]
쌍따옴표 없이도 채점되는 것 확인했습니다. 후처리 없이 그대로 제출해 주시면 됩니다.

**Reply by yms020615** [reply]
Q: It seems like quotation marks are required to test the data upon submission. Would it be possible to add them to the model output? I’m wondering if the model output needs to include quotation marks. It keeps generating captions without them.

A: I didn't quite understand the question. Could you explain it in more detail?

Q: When generating captions for submission through the model, sometimes captions are generated without quotation marks.
e.g.)
712e256c, "a red train on a platform at a station with a sign in the window of the front passenger door, and a sign in the window of the back passenger door"
1130bc2a, a man standing in front of a tv playing a bowling game on a Nintendo Wii video game console

It seems that all captions need to be in quotation marks for the submission to be recognized. I’m wondering if the model should generate captions with quotation marks, or if it’s acceptable to add them as a post-processing step.

A: I’ve confirmed that it’s graded without quotation marks. Please submit it as is, without any post-processing.

**Comment by BaconTomatoDeluxe** [comment]
주어진 데이터 시각화 자료에서 각 level의 데이터만을 분류할 수 있나요?
예를 들어 레벨 3의 데이터로만 이루어진 데이터셋을 만들 수 있나요?

**Reply by yms020615** [reply]
각 level에 존재하는 데이터에 대해서 저희는 제공하지 않고 크롤링의 방법도 불가능한 구조로 사이트가 구성되어 있습니다.

그러나 이미지를 하나하나 캡처하고 텍스트를 타이핑하는 방법으로 특정 level의 데이터셋을 모방한다면 저희는 막을 방법이 없으나, 그것이 최선의 방법으로 경량화된 데이터도 아닐 것입니다.

**Reply by yms020615** [reply]
Q: Is it possible to classify data by each level from the given data visualization materials? For example, could a dataset be created with only Level 3 data?

A: We do not provide data for each level, and the site is structured to prevent any crawling methods.

However, if you were to capture each image and type out the text manually to replicate a dataset for a specific level, we have no means to prevent it, though it wouldn’t be the most efficient approach and wouldn’t be a lightweight dataset either.

**Comment by strn18** [comment]
최종 결과(점수)는 어떻게 도출하나요? 제출한 csv 파일 중 private score가 가장 높은 것으로 산출하나요, 아니면 최종 제출한 코드를 바탕으로 거기서 나오는 csv 파일로 산출하나요?

**Reply by yms020615** [reply]
최종 **Reliability score**를 말씀하시는 것이라면, 제출해 주신 csv 파일 중 private score가 가장 높은 것으로 산출합니다. 최종 제출한 코드는 score의 재현성을 검증하기 위함입니다.

Q: How is the final result (score) determined? Is it based on the private score of the highest scoring submitted CSV file, or on the CSV file generated from the final submitted code?

A: If you are referring to the final **Reliability Score**, it is determined based on the highest private score among the submitted CSV files. The final submitted code is used to verify the reproducibility of the score.

**Comment by junhwan26** [comment]
앙상블 할 때 한 파일에서 앙상블 할 파일을 모두 불러와야하나요?

앙상블 할 모델의 결과값을 파일로 불러와도 가능한건가요?

**Reply by yms020615** [reply]
구현 과정에서는 어느 방법을 선택하여도 무방합니다.

후자를 선택하셨을 경우에는 실제 제출되는 코드에는 앙상블 할 파일이 만들어지는 과정을 포함하여야 합니다.

**Reply by yms020615** [reply]
Q: When ensembling, do I need to load all the files to be ensembled in one file?

Is it possible to load the results of the models to be ensembled from files?

A: Either method is acceptable during the implementation process.

If you choose the latter, the final submitted code must include the process for generating the files used in the ensemble.

**Reply by changbeendata** [reply]
혹시 Private Score가 최대인 submission이 아닌 다른 submission과 코드를 최종 결과물로 하여 발표도 해당 submission과 관련된 내용으로 진행하는 것이 가능할까요?

**Reply by yms020615** [reply]
Private Score가 최대인 submission이 아닌 다른 submission과 코드를 최종 결과물로 하는 것은 허용되지 않습니다. 그러나 발표에 다른 submission들(시행착오 등)에 대한 내용을 최대한 담고자 하는 것은 가능합니다.

**Reply by yms020615** [reply]
Q: Is it possible to use a submission and code other than the one with the highest Private Score as the final result and present content related to that submission?

A: It is not permitted to use a submission and code other than the one with the highest Private Score as the final result. However, you may include information about other submissions (such as trial and error) in the presentation as much as possible.
